javaslang/src/main/java/javaslang/collection/ArrayType.java

```diff
@@ -0,0 +1,147 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import java.io.Serializable;
+
+abstract class ArrayType<T> implements Serializable {
```

Comment:
I would love to see one Array abstraction on the file system, e.g. ArrayType and have the implementations as auxiliary classes. The ArrayType then needs to be added to the generator.

But I suggested below to call ObjectArrayType.xxx directly instead of using obj(). Maybe we need to call obj() then. Or do you have another suggestion?